### PR TITLE
Battering Ram - KNOCK KNOCK

### PR DESCRIPTION
**Pull request by DrDiasyl**
## About The Pull Request
Adds a unique item to the Security Armory - battering ram.
It can only be held with 2 arms and slows down by 1 point when held or put on the back.

Has enough force (20 * 3 demolition mod) to break through airlocks (similar to a fire axe), while doing so produces a loud noise and slightly shakes the screen 2 tiles from the airlock. 

You need to swing it by standing still for 0.5 seconds before hitting anything.

When hitting a person (and the person is standing and not anchored) - shove them away violently. Has a high chance of breaking bones. It still takes takes 0.5 to swing before hitting a person so its extremely easy to dodge.

If you try to hit an airlock when clumsy - you have a 30% chance of [slipping and falling backward](https://tenor.com/view/police-raid-fall-out-funny-gif-9719355) violently.

If you try to hit an airlock when it opens - you fall into it

### **VIDEOS UPDATED!**

<details>

https://github.com/user-attachments/assets/d706798c-a5d9-4a66-beff-9f1055519b08

https://github.com/user-attachments/assets/f1bb5139-18a1-4a1f-8e02-c7735945a181

https://github.com/user-attachments/assets/f2ffe0d3-3469-4843-a50f-0536a02b5c2d

image

image

</details>

## Why It's Good For The Game

Security the department that needed a breaching tool the most never got one. This gives them a thematic single tool to use in emergencies for breaking in during a Cult or something similar. Hacking, RCD, and Jaws of Life are much faster than ramming the airlock, but ramming is funny!

To use it Security still would need some cooperation with Engineering, Cargo, or Genetics to get insulated, as airlocks tend to get electrocuted when broken. 

Slowdown, swing time, obnoxiousness, and bulkiness deter it from being used as just only a kill stick or to tide in everywhere.
## Changelog
:cl: DrDiasyl aka DrTuxedo
add: Security has received a breaching tool - BATTERING RAM! Only one is found at Armory, great for breaking through airlocks, barricades, and ribs.
/:cl:

### CODE REVIEW

**Comment by Fikou** [comment]
why is it a subtype of the fire axe

**Comment by DrDiasyl** [comment]
> why is it a subtype of the fire axe

Its similar to it in functionality:

- Held with 2 hands
- Used for breaching

**Comment by carlarctg** [comment]
make it a tot obj as well like the mech remoavl tool

i loathe to say it but it might be a bit too slow - this is so much slower than a set of tools and a can of grey bull

**Comment by Fikou** [comment]
> > why is it a subtype of the fire axe
> 
> Its similar to it in functionality:
> 
> * Held with 2 hands
> * Used for breaching

those are very wide functions of an item and when you have to introduce new variables like "can this butcher" youre making stuff uglier for no real reason
also youre aware that this will oneshot all windows yes? you dont mention it anywhere in the pr

**Comment by DrDiasyl** [comment]
> > > why is it a subtype of the fire axe
> > 
> > 
> > Its similar to it in functionality:
> > 
> > * Held with 2 hands
> > * Used for breaching
> 
> those are very wide functions of an item and when you have to introduce new variables like "can this butcher" youre making stuff uglier for no real reason also youre aware that this will oneshot all windows yes? you dont mention it anywhere in the pr

Oh yeah, I forgot to tell that.

**Comment by JohnFulpWillard** [comment]
I've always wanted to add a ram for security, but I think such an item would require having a minimum of 2 people to operate, the slowness seems like an uncreative alternative in comparison.

**Comment by DrDiasyl** [comment]
> I've always wanted to add a ram for security, but I think such an item would require having a minimum of 2 people to operate, the slowness seems like an uncreative alternative in comparison.

I don't think there is any items that need 2 people to operate. Are there any that can be used as a baseline?
If no, I don't want to Code that from zero right now, as I find this PR finished.

**Comment by DrDiasyl** [comment]
> You can make it a "vehicle" to have 2 people holding it. then simply have them drive into a door
> 

Sounds like a hard recode from item to vehicle

**Comment by Loafin34** [comment]
i think what you could do is make a prompt appear when you swing at a door (use the offer code to check for people) and then if another person pushes the button on the screen, the battering ram slams into the door with more force. otherwise, this is just security fire axe.

**Comment by carlarctg** [comment]
also it shouldnt break windows instantly. fast yeah but leave the instant speed to the atmos techs

**Comment by DrDiasyl** [comment]
> i think what you could do is make a prompt appear when you swing at a door (use the offer code to check for people) and then if another person pushes the button on the screen, the battering ram slams into the door with more force. otherwise, this is just security fire axe.

Funny idea, I will consider it. But even this way it will remain quote on quote "security fire axe" as there is only 4 types of ways you can open a door:

- Break it
- Hack it
- Pry it
- Blow it

**Comment by Jacquerel** [comment]
The point of components is that you dont need to make things subtypes of random other objects in order to have similar behaviour. 
This shouldn't be a subtype of fire axe.

**Comment by DrDiasyl** [comment]
> The point of components is that you dont need to make things subtypes of random other objects in order to have similar behaviour. This shouldn't be a subtype of fire axe.

Alright, I will migrate it to its own type

**Comment by DaCoolBoss** [comment]
Sprite needs work, but that's a job for another PR.

Would it be reasonable to have this be an energy based weapon? Some kind of sci-fi piston that throws sparks everywhere when used and needs to go in a weapon charger after a few doors?

A do_after of about a second would be nice too. Not for balance reasons, just to give the feel of pulling the ram back, lining it up... then SMASH!ing it into the door.

**Comment by carlarctg** [comment]
image
opinions from resident sec tryhard. ignore or acknowledge at will

**Comment by SomeoneYouProbablyKnow** [comment]
I mean yea obviously. You can tryhard so much shit that's better than the ram. The ram is not something you go out of your way to get as sec 
It's something that if it turns out you need it later in the round you have access to without jumping through hoops (just make an officer or a greyshirt tank the shock).
It's not FOR John is i guess what i'm trying to say

The question is does security need a tool like that? one they get without trouble and without effort? Donno, i don't like security being so self sufficient, lets them keep their buissness "in house" with other windshielded people. But it is also a pretty cool way to do security access (even though it's not a limited amount of door moving shotgun shells, alas)

**Comment by norsvenska** [comment]
cursed people should have [this](https://youtu.be/TaTXpVECGxg?si=FawiZFDUI77Qy1n0) happen

**Comment by MrDoomBringer** [comment]
should definitely be insulated. if it requires insuls people will just hack a door 10x as fast

**Comment by SomeoneYouProbablyKnow** [comment]
> should definitely be insulated. if it requires insuls people will just hack a door 10x as fast

That would make it better than the fire axe at the fire axes job. Security shouldn't get those tools just for existing, it's already geared enough to steamroll anything on top of what departmental magpie secs take.
The game is so expensive there will always be a tool better at breaking doors than the ram. Shouldn't balance for those that powergame that much, they'd never use the ram unless it was op, hacking without insulation is faster than the ram, stealing an RCD is easier and more versetile, probably yelling at a nearby greyshit or engineer is better than the ram but that requires you to actually do something, the ram doesn't.
Besides you can always tank the shock, and that's okay for a tool they just HAVE without having to do anything.

**Comment by DaCoolBoss** [comment]
The other obvious way for sec to destroy a door, using laserguns, doesn't risk shock at all. I think it's fine for the ram to be shockproof for parity with that. Hacking or ion rifle/crowbar combo both risk shock but leave the door intact, so there's still a real choice to be made.

**Comment by DrDiasyl** [comment]
> The other obvious way for sec to destroy a door, using laserguns, doesn't risk shock at all. I think it's fine for the ram to be shockproof for parity with that. Hacking or ion rifle/crowbar combo both risk shock but leave the door intact, so there's still a real choice to be made.

Lasering takes away the ammo, to break a single airlock you need around 70%-ish of laser gun charge. And you really could've used that ammo to kill people, especially that you probably only have 1 gun.

While using a ram is infinite! With insulation.

**Comment by DaCoolBoss** [comment]
The ram being better than laser guns at busting down doors is a good thing though. Laserguns can do a bunch of stuff and the ram is just for this.

**Comment by SomeoneYouProbablyKnow** [comment]
> The ram being better than laser guns at busting down doors is a good thing though. Laserguns can do a bunch of stuff and the ram is just for this.

That's exactly why the ram is already better, you do not want to use up your laser gun and if you do you're welcome to do it, you'll be more vulnerable untill you charge it. Both opptions still weaker than a toolbelt an rcd or like a dozen other things and that's fine.

The ram doesn't need to be easier to use or faster than a laser gun nor does it need to be better than the actual fire axe. Security shouldn't just get all the tools and toys like that

**Comment by MrDoomBringer** [comment]
none of your arguments have provided a compelling reason for why we should have less specialized tools in the game.

At any rate we can fine tune the balance etc in a separate PR after this gets merged once we can empirically show that people aren't using this item because it's not good enough at the only job it has.

**Comment by ATH1909** [comment]
> > should definitely be insulated. if it requires insuls people will just hack a door 10x as fast
> 
> That would make it better than the fire axe at the fire axes job.

The fire axe doesn't require two hands at all times (like a potted plant) and slow you down while you're holding it.

I second the motion to make this not require insulation. Without that feature, you'll NEED to powergame (insulation) in order to use this ram for its intended purpose, which is jank as hell and would make this a noob trap.

**Comment by SkeletalElite** [comment]
any security officer who has insulated almost certainly already has a full set of tools which completely defeats the purpose of this thing existing.

**Comment by GuillaumePrata** [comment]
> Instantly breaks windows and grilles

I think it is smarter to keep this as only an anti airlock tool.

If it instantly breaks windows, it will be often faster to go through them than to use it on the airlocks as 90% of the rooms in the game have windows.

Now if it is designed only for airlocks, we can give it a lot more power.
Personally, I would make it work like the watertank backpacks, something you carry on your back with a minor slowdown and with an insulated extension you can deploy to your hand to smash airlocks safely and fast.

This will give it a massive downside of losing your backpack/MODsuit, so it won't trivialize things like Malf AI sat break-ins and will also push security to work together and cover the breach officer as they are heavily neutered without a backpack+slowdown.

Good for coopecurity, risky for solo protagcurity and basically unusable by antags/random cremembers.

**Comment by FlamingIceGolem** [comment]
Is it possible to have the Security Gas Mask HALT-feature say "SECURITY, OPEN UP!!!", when you're wielding this tool

**Comment by DrDiasyl** [comment]
> Is it possible to have the Security Gas Mask HALT-feature say "SECURITY, OPEN UP!!!", when you're wielding this tool

This is brilliant, sadly I have no idea who the hell is the voice actor of Sec hailer, as such I will have to record it myself and it might sound much different from other hailers lol

**Comment by DrDiasyl** [comment]
UPDATED:

- No longer a subtype of fire axe, no longer instantly breaks windows
- Now you need to swing it (0.5 seconds) before hitting anything
- Force is now 20 and demolition mod is 3 (9~ hits to break a not-reinforced airlock)
- If trying to hit an opening/open airlock - you fall into it

**Comment by Loafin34** [comment]
i think the demolition modifier should be higher. this is comparable to big slappy, which does 30 damage to structures (which also doesnt trigger it's knockback), and will take around 20 seconds to break in a door for a backpack-only item, which is slow enough for people to escape.

**Comment by DrDiasyl** [comment]
> i think the demolition modifier should be higher. this is comparable to big slappy, which does 30 damage to structures (which also doesnt trigger it's knockback), and will take around 20 seconds to break in a door for a backpack-only item, which is slow enough for people to escape.

20 * 3 = 60 damage.
It takes 15 seconds to destroy an not-reinforced airlock.

**Comment by CRITAWAKETS** [comment]
This should probably be made to repeat upon being used on an airlock just so you don't have to click like 10 times as a convenience feature.

**Comment by projectkepler-ru** [comment]
> I've always wanted to add a ram for security, but I think such an item would require having a minimum of 2 people to operate, the slowness seems like an uncreative alternative in comparison.

having played skyrat with the two people required battering ram it just made it so it *never* gets used, dead serious
that's a bad idea

File: code/game/objects/items/battering_ram.dm [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+/obj/item/batteringram
+	icon = 'icons/obj/weapons/fireaxe.dmi'
+	icon_state = "battering_ram0"
+	base_icon_state = "battering_ram"
+	lefthand_file = 'icons/mob/inhands/weapons/axes_lefthand.dmi'
+	righthand_file = 'icons/mob/inhands/weapons/axes_righthand.dmi'
+	name = "battering ram"
+	desc = "An extremely heavy and unwieldy hunk of metal used to break down airlocks. Why would anyone ram a sliding airlock?"
+
+	w_class = WEIGHT_CLASS_HUGE
```
Comment (projectkepler-ru) [inline code/game/objects/items/battering_ram.dm:15]: Given carlarc suggested it should be a traitor objective, maybe make it bulky?

File: code/game/objects/items/battering_ram.dm [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+	item_flags = SLOWS_WHILE_IN_HAND | IMMUTABLE_SLOW
+	slot_flags = ITEM_SLOT_BACK
+	obj_flags = CONDUCTS_ELECTRICITY
+	resistance_flags = FIRE_PROOF
+	armor_type = /datum/armor/item_batteringram
+
+	demolition_mod = 3 // 9 hits to down a airlock
```
Comment (projectkepler-ru) [inline code/game/objects/items/battering_ram.dm:22]: 9 hits to down an airlock for a weapon that is owned by one person, is not replacable if it was stolen

rough

File: code/game/objects/items/battering_ram.dm (hunk 1 above)
Comment (Fikou) [reply code/game/objects/items/battering_ram.dm:15]: ok but its not one?

File: code/game/objects/items/battering_ram.dm (hunk 2 above)
Comment (Fikou) [reply code/game/objects/items/battering_ram.dm:22]: this isnt really a code review

File: code/game/objects/structures/fireaxe.dm [hunk 3]
```diff
@@ -20,6 +20,8 @@
 	var/item_overlay = "axe"
 	/// Whether we should populate our own contents on Initialize()
 	var/populate_contents = TRUE
+	/// Is the held_item must be unwielded?
```
Comment (Fikou) [inline code/game/objects/structures/fireaxe.dm:23]: spellcheck this

File: sound/effects/bang.ogg
Comment (Sadboysuss) [inline sound/effects/bang.ogg:1]: what did you change?

File: code/game/objects/items/battering_ram.dm [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+		living_user.Knockdown(3 SECONDS)
+		living_user.safe_throw_at(throwtarget, 1, 1, force = MOVE_FORCE_STRONG)
+
+		to_chat(living_user, span_userdanger("You [pick(attack_verb_simple)] [target], but slip due inertia!"))
+		visible_message(span_warning("[living_user] slips backwards!"))
+		playsound(living_user, 'sound/misc/slip.ogg', 100)
+
+/obj/item/batteringram/suicide_act(mob/living/user)
+	user.visible_message(span_suicide("[user] [pick(attack_verb_continuous)] [user.p_them()]self open! It looks like [user.p_theyre()] trying to commit suicide!"))
+	playsound(user, hitsound, 100, ignore_walls = FALSE)
```
Comment (Sadboysuss) [inline code/game/objects/items/battering_ram.dm:116]: make this pressure affected please

File: code/game/objects/items/battering_ram.dm [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+	slowdown = 1
+	drag_slowdown = 1.5 // So you cannot circumvent the slowdown
+	throw_range = 2
+
+	attack_verb_continuous = list("slams", "breaks", "demolishes", "rams", "batters", "breaches")
+	attack_verb_simple = list("slam", "break", "demolish", "ram", "batter", "breach")
+
+	hitsound = 'sound/effects/bang.ogg'
+	pickup_sound = 'sound/items/handling/heavy_pickup.ogg'
+	drop_sound = 'sound/items/handling/heavy_drop.ogg'
```
Comment (Sadboysuss) [inline code/game/objects/items/battering_ram.dm:36]: maybe put `sound_vary = TRUE` on this too

**Comment by nukularpower** [comment]
I dont know why none of the comments mention this but the issue I have with this personally is the whole "violently shoves them away and breaks bones" bit when used on a player.   It sounds like you could just get a stabilized red to deal with slowdown and use this as an uber valid tool, if hitting someone with it floors them and breaks their bones.   None of the videos show it but it sounds like pre-nerf version of the arm implant, which was broken as hell.

Also, will the TG trend of overbuffing sec in general ever stop?  The guy above me that said sec shouldnt just get all the tools was spot on

**Comment by DrDiasyl** [comment]
> I dont know why none of the comments mention this but the issue I have with this personally is the whole "violently shoves them away and breaks bones" bit when used on a player. It sounds like you could just get a stabilized red slime to deal with slowdown and use this as an uber valid tool, if hitting someone with it floors them and breaks their bones. None of the videos show it but it sounds like pre-nerf version of the strongarm implant, which was broken as hell, except worse cuz it's also breaking bones.
> 
> The guy above me that said sec shouldnt just get all the tools was spot on. If all this can do is slowly open doors, fine - but if it also insta breaks windows AND is a powergamer weapon, not a fan at all. Personally feel it should do basically 0 damage unless you like, charge into someone, but doubt a charge thing was coded into it.

It doesn't shove people who are lying down.

You STILL have to SWING for HALF A SECOND even when hitting a PERSON and not a door, so it's very easy to dodge.

**Comment by DrDiasyl** [comment]
Also baton is still the most superior weapon in the game nothing can top it.

**Comment by nukularpower** [comment]
I dont see batons being ridiculous as a reason that would justify something ELSE that could insta-win fights if you get that .5 seconds off - that isnt so long that it would be impossible, and breaking bones is obviously strong even wtihout taking the KO into account.  And regardless of how "good" it is, I'm sure there are quite a few that would RATHER break bones than capture anyways, or just include that as part of the process.  Doesnt seem like it should be usable in combat at all to me, at any rate

**Comment by SomeoneYouProbablyKnow** [comment]
I think the do_after is enough to make sure it's a bad weapon. Very funny execution tool though. I honestly don't think the ability to abuse something with xenobio should have much bearing on the balance of items. Mostly because fuck xenobio and also because it frankly does not matter.

I do maintain that giving security a discount fireaxe is not ideal and (even though it's not gonna be used by powergamers) make security even more needlessly self sufficient and isolationist.

**Comment by githubuser4141** [comment]
> > > > why is it a subtype of the fire axe
> > > 
> > > 
> > > Its similar to it in functionality:
> > > 
> > > * Held with 2 hands
> > > * Used for breaching
> > 
> > 
> > those are very wide functions of an item and when you have to introduce new variables like "can this butcher" youre making stuff uglier for no real reason also youre aware that this will oneshot all windows yes? you dont mention it anywhere in the pr
> 
> if it being able to break window easily is the concern maybe we need to re-do how demolition mod is done? because atm its also shared with mech and borg so a breaching slug will effortlessly break those down too

yogsstation server has a max cap on mechs, think it's like 2x demo mod MAX on mechs, so having a 10x demo mod multipler won't grant you the power to oneshot a fully armored durand

**Comment by Swiftkor** [comment]
As a sec main who enjoys using tools that aren't always optimal but are more flavorful, I really don't see myself using this. I agree that sec has needed a breaching tool for a long time, but I doubt I'd ever use something that slowly breaches doors, gives slowdown, and isn't even insulated. Especially since the hydraulic clamp that comes on paddy is getting it's ability to wrench open doors fixed in [#85514](https://github.com/tgstation/tgstation/pull/85514).

Only way I can see this being worth using is if it gets insulation. Maybe making this a paddy attachment that's available in sec's mech bay would differentiate it from the fireaxe and give the paddy more utility as a breacher mech.

**Comment by DrDiasyl** [comment]
> As a sec main who enjoys using tools that aren't always optimal but are more flavorful, I really don't see myself using this. I agree that sec has needed a breaching tool for a long time, but I doubt I'd ever use something that slowly breaches doors, gives slowdown, and isn't even insulated. Especially since the hydraulic clamp that comes on paddy is getting it's ability to wrench open doors fixed in [#85514](https://github.com/tgstation/tgstation/pull/85514).
> 
> Only way I can see this being worth using is if it gets insulation. Maybe making this a paddy attachment that's available in sec's mech bay would differentiate it from the fireaxe and give the paddy more utility as a breacher mech.

I don't want to imbalance it in any way.
If this will be found not very used - anyone can add insulation themselves post factum.

Also I've been thinking about making it possible to attach it onto Paddy/Segway to carry around quicker.

**Comment by mkvrs** [comment]
noob trap item

**Comment by DrDiasyl** [comment]
Later.

**Pull request closed without merging**